dataflow/gemma/custom_model_gemma.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from typing import Any
+from typing import Dict
```

Comment:
As per [PEP-585](https://peps.python.org/pep-0585):

- `typing.Dict` can be replaced by just `dict`
- `Iterable` should be imported from `collections.abc` instead of `typing`
- `Optional` can be replaced with `|` types, like `int | None` instead of `Optional[int]`
- `Sequence` should be imported from `collections.abc` instead of `typing`

Apache Beam might have typing issues with that, but adding `from __future__ import annotations` should fix it.